### ISSUE TITLE
[Request] Add support for multi-value parameters

### ISSUE THREAD

**Issue by kbrownlees** [dougmoscrop/serverless-http#66]
Just want to raise this so it is here for future reference, this is the same as https://github.com/awslabs/aws-serverless-express/issues/196. Application load balancer supports this via a configuration option in the backend, and api gateway has recently got support for it. Pretty nasty but unfortunately pretty common in a lot of codebases these days.

For application load balancer this is an example event:
```
multiValueHeaders:  {
   accept: [ 'application/json, text/javascript, */*; q=0.01' ],
  'accept-encoding': [ 'gzip, deflate, br' ],
  'accept-language': [ 'en-GB,en-US;q=0.9,en;q=0.8' ],
  ...
}
```
```
multiValueQueryStringParameters: {
  'dateRange%5B%5D': [ '2018-12-14', '2018-12-14' ],
  roleId: [ '0' ]
}
```

The behaviour of my express app currently with that same request is:

```
req.query == {
  'dateRange': ['2018-12-14', '2018-12-14' ],
  'roleId': '0'
}
```

Not entirely sure how you would go about handling this in a useful manner.

I would probably be happy if there was some way of way of pre-processing the event so I have a hook to transform the data to ensure that ServerlessRequest behaves itself.

**Comment by jamesvillarrubia** [comment]
I just submitted a PR for this. https://github.com/dougmoscrop/serverless-http/pull/67 Hopefully it'll roll out soon.  In the meantime, you can fiddle with a field overwrite ahead of the handler call in AWS.

```javascript
//index.js
app.setup(server)

module.exports = {
  handler: async (event, context) => {
    event.queryStingParameters = event.multiValueQueryStringParameters
    return server(event, context)
  }
}
```

**Comment by kbrownlees** [comment]
Thanks @jamesvillarrubia for pointing me in the right direction.

Note for anyone trying to do this (for application load balancer at least) you need to convert the querystrings / headers on the way in and the headers on the way out.

eg:
```coffeescript
module.exports = {
    handler: (event, context) =>
        event = util.serverlessTransformMultiValuesIncoming(event)
        event = util.serverlessCleanQueryStringParameters(event)

        server(event, context)
        .then (res) =>
            res = util.serverlessTransformMultiValuesOutgoing(res, event)
            res
}
```

Keys to transform inbound are: `multiValueQueryStringParameters->queryStringParameters` and `multiValueHeaders->headers`. Outbound there are only headers so `headers->multiValueHeaders`

**Comment by dougmoscrop** [comment]
Released in 1.9.0
